pkg/cloudcommon/grpc.go

```diff
@@ -127,3 +131,65 @@ var GRPCClientKeepaliveParams = keepalive.ClientParameters{
 var GRPCServerKeepaliveEnforcement = keepalive.EnforcementPolicy{
 	MinTime: 1 * time.Second,
 }
+
+// GRPCError unwraps status.Status errors, this avoids
+// ugly type print outs when various library code converts
+// the error to a string.
+func GRPCErrorUnwrap(err error) error {
+	st, ok := status.FromError(err)
+	if ok && st != nil {
+		return errors.New(st.Message())
+	}
+	return err
+}
+
+type GRPCStreamRecv[Object any] interface {
```

Comment:
I thought we'd be able to not use generics for a bit longer :)

Reply:
They're actually pretty useful :) It's either this or I need to define some new protogen options and auto-generate this code from the protobufs, which means I basically write this same code with templates instead of generics. I actually prefer the generics as VSCode understands them and can detect syntax/compile errors.

The only drawback of generics is you can't use generic functions on an object, which is somewhat limiting. But the object can be generic.